src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -1643,3 +1646,232 @@ def test_rename_symbol():
+           subroutine.symbol_table.lookup('b_mod')
+    # not_used1 and not_used2 exist and are DataSymbols now
+    assert isinstance(not_used1, DataSymbol)
+    assert isinstance(subroutine.symbol_table.lookup('not_used2'), DataSymbol)
+
+    # a_2 is not yet resolved because if is from another symbol table,
+    # resolve that symbol table too
+    assert not isinstance(a_2, DataSymbol)
+    subroutine.parent.symbol_table.resolve_imports()
+    # In this case check that the visibility stays PRIVATE
+    assert isinstance(a_2, DataSymbol)
+    assert a_2.visibility == Symbol.Visibility.PRIVATE
+
+    # Clean up the config instance
+    Config._instance = None
+
+
+def test_resolve_imports_name_clashes(fortran_reader, tmpdir):
+    ''' Tests the SymbolTable resolve_imports method raises the appropriate
+    errors when it find name clashes. '''
+
+    with open(os.path.join(str(tmpdir), "a_mod.f90"), "w") as module:
+        module.write('''
+        module a_mod
+            integer :: name_clash1
+        end module a_mod
+        ''')
+    with open(os.path.join(str(tmpdir), "b_mod.f90"), "w") as module:
+        module.write('''
+        module b_mod
+            integer :: not_a_name_clash
+            integer :: name_clash1
+            integer :: name_clash2
+            private not_a_name_clash
+        end module b_mod
+        ''')
+    psyir = fortran_reader.psyir_from_source('''
+        module test_mod
+            contains
+            subroutine test()
+                use a_mod
+                use b_mod
+                integer :: not_a_name_clash ! because its private in the module
+                integer :: name_clash2
+
+                name_clash1 = name_clash2 + not_a_name_clash
+            end subroutine test
+        end module test_mod
+    ''')
+    subroutine = psyir.walk(Routine)[0]
+    symtab = subroutine.symbol_table
+
+    # Set up include_path to import the proper modules
+    Config.get()._include_paths = [str(tmpdir)]
+
+    with pytest.raises(SymbolError) as err:
+        symtab.resolve_imports([symtab.lookup('b_mod')])
+    assert ("Found a name clash with symbol 'name_clash2' when importing "
+            "symbols from container 'b_mod'." in str(err.value))
+
+    with pytest.raises(SymbolError) as err:
+        symtab.resolve_imports([symtab.lookup('a_mod')])
+    assert ("Found a name clash with symbol 'name_clash1' when importing "
+            "symbols from container 'a_mod', this symbol was already defined "
+            "in 'b_mod'." in str(err.value))
+
+    # Clean up the config instance
+    Config._instance = None
+
+
+def test_resolve_imports_with_datatypes(fortran_reader, tmpdir):
+    ''' Tests the SymbolTable resolve_imports method. '''
+    with open(os.path.join(str(tmpdir), "my_mod.f90"), "w") as module:
+        module.write('''
+        module my_mod
+            type my_type
+                integer :: field
+                integer, dimension(:,:) :: array
```

Comment:
I think `array` must either be allocatable or have fixed dimensions.

Reply:
Added fixed dimensions, and since I was at it, I now assert that this dimension value is seen in the shape of the local symbol after resolving the import.